DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/ip_authoring_mmhost/README.md

```diff
@@ -0,0 +1,368 @@
+    64,      // data width
+    16,      // latency
+    1,       // read_write_mode, 0: ReadWrite, 1: Read, 2: Write
+    1,       // maxburst
+    0,       // align, 0 defaults to alignment of the type
+    1        // waitrequest, 0: false, 1: true
+  ) int *my_pointer;
+
+  void operator()() const {
+    ...
+    // Kernel code
+    my_pointer[x] = y;
+    ...
+  }
+}
+```
+
+
+
+An Avalaon bus is generate for every unique global memory in the system. 
+
+If the mmhost argument is specified as a conduit interface then an input wire is generated at the top-level device image IP to carry the the pointer address into the kernal. 
+
+If the mmhost argument is specified as a register-mapped argument, the input pointer is written into a register map associated with the kernal. An Avalon bus is generated for writing those arguments into a register map, with no seperate wire needed at the top level device image to input the address.
+#### Example 2: Changing default mmhost interface. 
+```c++
+// A memory-mapped interface that is implemented as a register.
+  register_map_mmhost(
+    1,       // buffer_location or aspace
+    28,      // address width
+    64,      // data width
+    16,      // latency
+    1,       // read_write_mode, 0: ReadWrite, 1: Read, 2: Write
+    1,       // maxburst
+    0,       // align, 0 defaults to alignment of the type
+    1        // waitrequest, 0: false, 1: true
+  ) int *memory_mapped_pointer;
+// A memory-mapped interface that is implemented as a wire.
+  conduit_mmhost(
+    1,       // buffer_location or aspace
+    28,      // address width
+    64,      // data width
+    16,      // latency
+    1,       // read_write_mode, 0: ReadWrite, 1: Read, 2: Write
+    1,       // maxburst
+    0,       // align, 0 defaults to alignment of the type
+    1        // waitrequest, 0: false, 1: true
+  ) int *memory_mapped_pointer;
+```
+
+If unspecified, when using only the ```mmhost(...)``` macro, a register_map based input argument interface is used.
+
+### Memory Base Addresses assigned by the compiler
+
+The compiler infers a unique global memory for each pointer argument that is annotated with a buffer location. An entry describing the interface of each such memory can be found in the file ipinterfaces.xml which is generated in the "*.prj" output directory. The entry specifies various properties of the memory including the start address that was assigned to the memory by the compiler. Each buffer_location will have its address defined there, among other parameters. 
+
+To understand how these start addresses are generated and how the compiler assigns special values to the top bits of the addresses please see the complete documentation on memory-mapped interfaces. If the parameters are not set correctly, this can lead to undefined behaviour.
+
+The file will have the following entries based on the kernel arguments used in this tutorial:
+
+Example of a pointer argument in the ipinterfaces.xml:
```

Comment:
Replace this description by stating that the information about the start address and other properties can be found in the Memory Viewer.